lib/avromatic/model/attributes.rb

```diff
@@ -73,7 +87,10 @@ def initialize(data = {})
         end
 
         unless Avromatic.allow_unknown_attributes || valid_keys.size == data.size
-          raise ArgumentError.new("Unexpected attributes for #{self.class.name}: #{(data.keys - valid_keys).map(&:to_s).join(', ')}. Complete arguments: #{data}")
+          unknown_attributes = data.keys.map(&:to_s) - valid_keys.map(&:to_s)
+          message = "Unexpected arguments for #{self.class.name}#initialize: #{unknown_attributes.join(', ')}. " \
+            "Provided arguments: #{data.inspect}"
+          raise Avromatic::Model::UnknownAttributeError.new(message, unknown_attributes)
```

Comment:
Can you also include the expected attributes in the `message` and pass them through to `UnknownAttributeError`?